planner/core/planbuilder.go

```diff
@@ -1394,38 +1394,23 @@ func (b *PlanBuilder) buildShow(ctx context.Context, show *ast.ShowStmt) (Plan,
 			return nil, err
 		}
 	}
-	if np != mockTablePlan {
-		fieldsLen := len(mockTablePlan.schema.Columns)
+	if np != p {
+		fieldsLen := len(p.schema.Columns)
 		proj := LogicalProjection{Exprs: make([]expression.Expression, 0, fieldsLen)}.Init(b.ctx)
 		schema := expression.NewSchema(make([]*expression.Column, 0, fieldsLen)...)
-		for _, col := range mockTablePlan.schema.Columns {
+		for _, col := range p.schema.Columns {
 			proj.Exprs = append(proj.Exprs, col)
 			newCol := col.Clone().(*expression.Column)
 			newCol.UniqueID = b.ctx.GetSessionVars().AllocPlanColumnID()
 			schema.Append(newCol)
 		}
 		proj.SetSchema(schema)
 		proj.SetChildren(np)
-		physical, err := DoOptimize(ctx, b.optFlag|flagEliminateProjection, proj)
-		if err != nil {
-			return nil, err
-		}
-		return substitutePlaceHolderDual(physical, p), nil
+		return proj, nil
```

Comment:
Do we need to eliminate projection here?

Reply:
Prvious one is not a valid project elimination.
For now, show cannot remove any of its output column.
So the project should be remained.

Reply:
> Prvious one is not a valid project elimination.
For now, show cannot remove any of its output column.
So the project should be remained.

Sorry I didn't understand this, please explain a little bit more. Also, shall we set `flagEliminateProjection` of `b.optFlag` here?

Reply:
Oh, sorry i misread the comment.
We can add flag here.
But the project in the that test won't be removed